src/fsharp/TypeChecker.fs

```diff
@@ -2112,9 +2112,9 @@ module GeneralizationHelpers =
             match tp.Constraints |> List.partition (function (TyparConstraint.CoercesTo _) -> true | _ -> false) with 
             | [TyparConstraint.CoercesTo(cxty,_)], others -> 
                  // Throw away null constraints if they are implied 
-                 match others |> List.filter (function (TyparConstraint.SupportsNull(_)) -> not (TypeSatisfiesNullConstraint cenv.g m cxty) | _ -> true) with
-                 | [] -> Some cxty
-                 | _ -> None
+                 if others |> List.exists (function (TyparConstraint.SupportsNull(_)) -> not (TypeSatisfiesNullConstraint cenv.g m cxty) | _ -> true) 
```

Comment:
@smoothdeveloper Per CONTRIBUTING,md, that sort of thing would be a separate cleanup PR (though it's fair to say that reducing line length isn't really a cleanup aim at the moment)